### ISSUE TITLE
onFinishChange Event

### ISSUE THREAD

**Issue by poroia** [cocopon/tweakpane#227]
Hey! Just wondering there is / will be support for a "finishChange" kind of event for inputs like in `dat.gui`? I personally need it for computationally expensive tweaks.

Thanks for the amazing library btw, it's super aesthetic and has so many great features!

**Comment by hirohe** [comment]
sounds like `debounce` the change event ?

**Comment by poroia** [comment]
I believe so. If you had a number input, then on mouseUp of the slider for it, the object would be modified. Hope that makes sense.

**Comment by cocopon** [comment]
Thank you for using Tweakpane!

Should it be fired even for discrete operations (e.g. changing a text, toggling a checkbox, etc.)?

**Comment by technbuzz** [comment]
I would like to use it with sliders

**Comment by poroia** [comment]
> Should it be fired even for discrete operations (e.g. changing a text, toggling a checkbox, etc.)?

Just checked `dat.gui` and they fire the onFinishChange event on discrete operations, too.

**Comment by technbuzz** [comment]
I think for checkbox, onFinishChange will wait for how many times box is
checked and unchecked in specified time frame?

**Comment by poroia** [comment]
> I think for checkbox, onFinishChange will wait for how many times box is checked and unchecked in specified time frame?

Hmm, that sounds pretty interesting, but I don't think such a throttle is needed as it sounds like a pretty specific use case.

**Comment by cocopon** [comment]
> I don't think such a throttle is needed as it sounds like a pretty specific use case

I agree, it may be out of scope.

**Comment by cocopon** [comment]
I think that `change` handler and `finishchange` handler have similar logic, so how about this style?:

```js
input.on('change', (ev) => {
  if (ev.last) {
    // finishchange
  } else {
    // change
  }
});
```

**Comment by poroia** [comment]
Sorry for the late response, but yeah, that looks good!

**Comment by vujadin** [comment]
Any news on this one? I'm working on history implementation in my app (undo/redo) and I absolutely need onFinishChange event...

**Comment by cocopon** [comment]
Released in 3.0.3, thank you for reporting the issue.
